File: app/src/main/java/com/frankegan/verdant/VerdantApp.java
package com.frankegan.verdant;

import android.app.Application;
import android.content.Context;
import android.content.ContextWrapper;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;
import com.pixplicity.easyprefs.library.Prefs;

/**
 * @author frankegan created on 6/2/15.
 */
public class VerdantApp extends Application {

    private static VerdantApp instance;
    private static RequestQueue mRequestQueue;

    public VerdantApp() {
        instance = this;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mRequestQueue = Volley.newRequestQueue(this);
        // Initialize the Prefs class
        new Prefs.Builder()
                .setContext(this)
                .setMode(ContextWrapper.MODE_PRIVATE)
                .setPrefsName(getPackageName())
                .setUseDefaultSharedPreference(true)
                .build();
    }

    public static Context getContext() {
        return instance;
    }

    public static RequestQueue getVolleyRequestQueue(){
        return mRequestQueue;
    }
}

File: app/src/main/java/com/frankegan/verdant/VerdantApp.kt
package com.frankegan.verdant

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build
import androidx.core.content.systemService


/**
 * @author frankegan created on 6/2/15.
 */
class VerdantApp : Application() {

    companion object {

        @JvmStatic
        lateinit var instance: VerdantApp
            private set

        const val DOWNLOAD_CHANNEL_ID = "download_channel_is"
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
        initChannels()
    }

    private fun initChannels() {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) return
        /// Create the NotificationChannel, but only on API 26+ because
        // the NotificationChannel class is new and not in the support library
        val name = getString(R.string.download_channel_name)
        val desc = getString(R.string.download_channel_desc)
        val importance = NotificationManager.IMPORTANCE_DEFAULT
        val channel = NotificationChannel(DOWNLOAD_CHANNEL_ID, name, importance).apply { description = desc }
        // Register the channel with the system; you can't change the importance
        // or other notification behaviors after this
        systemService<NotificationManager>().createNotificationChannel(channel)
    }
}